### PR TITLE
Added zbar recipe, first revision.

### PR DESCRIPTION
**Pull request by barctor**
Adds a recipe for [zbar](http://zbar.sourceforge.net/) providing:
* Main package: cmd:zbarimg - command line program to scan QR codes & barcodes in image files
* Devel package: lib:libzbar - C library/headers exposing API to scan barcodes & QR codes

Currently builds package without GTK, QT, or Python support.

### CODE REVIEW

**Comment by extrowerk** [comment]
Not tested, but looks good.

**Comment by Begasus** [comment]
Latest release is 0.23.90 according to repology: https://repology.org/project/zbar/versions
Hosted at: https://github.com/mchehab/zbar/

**Comment by barctor** [comment]
Thanks @Begasus I didn't know development had moved to github.
I've amended the commit to use the latest release from the linked git repo.
As far as tests go:
* Main package: zbarimg is able to scan images and print any QR code data found
* Devel package: i can build a project that references zbar.h but have yet to test the actual API to see if it works as intended

File: media-gfx/zbar/zbar-0.23.90.recipe [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+libVersion="0.3.0"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+commandSuffix=$secondaryArchSuffix
+commandBinDir=$binDir
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	commandSuffix=
+	commandBinDir=$prefix/bin
+fi
```
Comment (Begasus) [inline media-gfx/zbar/zbar-0.23.90.recipe:30]: libVersion comes after commandSuffix: see https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#ordering

File: media-gfx/zbar/zbar-0.23.90.recipe [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	lib:libzbar$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libMagick++_7.Q16HDRI$secondaryArchSuffix
+	lib:libMagickCore_7.Q16HDRI$secondaryArchSuffix
+	lib:libMagickWand_7.Q16HDRI$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libjpeg$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
```
Comment (Begasus) [inline media-gfx/zbar/zbar-0.23.90.recipe:44]: Sort alphabetically (also in BUILD_REQUIRES)

File: media-gfx/zbar/zbar-0.23.90.recipe [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+BUILD_REQUIRES="
+	imagemagick${secondaryArchSuffix}_devel
+	devel:libMagick++_7.Q16HDRI$secondaryArchSuffix
+	devel:libMagickCore_7.Q16HDRI$secondaryArchSuffix
+	devel:libMagickWand_7.Q16HDRI$secondaryArchSuffix
+	devel:libiconv$secondaryArchSuffix
+	devel:libjpeg$secondaryArchSuffix
+	devel:libintl$secondaryArchSuffix
+	haiku${secondaryArchSuffix}_devel
```
Comment (Begasus) [inline media-gfx/zbar/zbar-0.23.90.recipe:64]: Changes:
```diff
- 	imagemagick${secondaryArchSuffix}_devel (not needed)
+	haiku${secondaryArchSuffix}_devel
	devel:libMagick++_7.Q16HDRI$secondaryArchSuffix
	devel:libMagickCore_7.Q16HDRI$secondaryArchSuffix
	devel:libMagickWand_7.Q16HDRI$secondaryArchSuffix
	devel:libiconv$secondaryArchSuffix
	devel:libjpeg$secondaryArchSuffix
	devel:libintl$secondaryArchSuffix
-	haiku${secondaryArchSuffix}_devel
```

File: media-gfx/zbar/zbar-0.23.90.recipe [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+BUILD_PREREQUIRES="
+	cmd:aclocal
+	cmd:autoconf
+	cmd:automake
+	cmd:autopoint
+	cmd:awk
+	cmd:find
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:libtool
```
Comment (Begasus) [inline media-gfx/zbar/zbar-0.23.90.recipe:75]: Change for secondary architecture:
```diff
- 	cmd:libtool
+ 	cmd:libtoolize$secondaryArchSuffix
```

File: media-gfx/zbar/zbar-0.23.90.recipe [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+	"
+
+defineDebugInfoPackage zbar$secondaryArchSuffix \
+	"$commandBinDir"/zbarimg \
+	"$libDir"/libzbar.so.$libVersion
+
+BUILD()
+{
+	autoreconf -vfi
+	LDFLAGS="-lintl" runConfigure ./configure \
```
Comment (Begasus) [inline media-gfx/zbar/zbar-0.23.90.recipe:88]: We need to move the binary to $prefix/bin here too:
```diff
- 	LDFLAGS="-lintl" runConfigure ./configure \
+ 	LDFLAGS="-lintl" runConfigure --omit-dirs binDir ./configure \
+ 	 	--bindir=$commandBinDir \

**Review by Begasus — CHANGES_REQUESTED** [review]

**Comment by barctor** [comment]
Thanks for the review @Begasus 
I think I have addressed all the issues you found in the recipe.
I'll also make sure I read through the guide more thoroughly before submitting another recipe :)

File: media-gfx/zbar/zbar-0.23.90.recipe [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+	lib:libMagick++_7.Q16HDRI$secondaryArchSuffix
+	lib:libMagickCore_7.Q16HDRI$secondaryArchSuffix
+	lib:libMagickWand_7.Q16HDRI$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	zbar${secondaryArchSuffix}_devel = $portVersion
+	devel:libzbar$secondaryArchSuffix = $libVersionCompat
+	"
+
```
Comment (Begasus) [inline media-gfx/zbar/zbar-0.23.90.recipe:52]: No new line between PROVIDES_devel and REQUIRES_devel
```diff
-
```

**Comment by barctor** [comment]
Sorted the PROVIDES_devel and REQUIRES_devel spacing, and removed the extra space between libVersion and PROVIDES

**Review by Begasus — APPROVED** [review]

**Comment by Begasus** [comment]
Thanks! :+1:

**Pull request merged**